### PR TITLE
Ordere v3: remove sys chan: clean consensus

### PR DESCRIPTION
**Pull request by tock-ibm**
Change-Id: I600362026a64d792782b3bf9d4037348821a4a4e

#### Type of change
- Improvement (improvement to code, performance, etc)

#### Description

Ordere v3: remove sys chan: clean consensus, remove InactiveChainRegistry from consenter
Remove inactive.Chain

#### Related issues

Issue: #4185

### CODE REVIEW

File: orderer/consensus/etcdraft/consenter.go [hunk 1]
```diff
@@ -232,19 +208,10 @@ func (c *Consenter) HandleChain(support consensus.ConsenterSupport, metadata *co
...
-		haltCallback = func() {
-			c.InactiveChainRegistry.TrackChain(support.ChannelID(), nil, func() { c.ChainManager.CreateChain(support.ChannelID()) })
-			c.ChainManager.ReportConsensusRelationAndStatusMetrics(support.ChannelID(), types.ConsensusRelationConfigTracker, types.StatusInactive)
-		}
-	} else {
-		// when we do NOT have a system channel, we switch to a follower.Chain upon eviction.
-		c.Logger.Info("Without system channel: after eviction Registrar.SwitchToFollower will be called")
-		haltCallback = func() { c.ChainManager.SwitchChainToFollower(support.ChannelID()) }
-	}
+	// called after the etcdraft.Chain halts when it detects eviction form the cluster.
```
Comment (yacovm) [inline orderer/consensus/etcdraft/consenter.go]: form-->from

File: orderer/consensus/etcdraft/consenter.go [hunk 2]
```diff
@@ -232,19 +208,10 @@ func (c *Consenter) HandleChain(support consensus.ConsenterSupport, metadata *co
...
+	// when we do NOT have a system channel, we switch to a follower.Chain upon eviction.
+	c.Logger.Info("Without system channel: after eviction Registrar.SwitchToFollower will be called")
```
Comment (yacovm) [inline orderer/consensus/etcdraft/consenter.go]: why would we want to mention the system channel term in logs? and I'm also not sure how informative this message is. Can we instead explain what is going on?

Comment (tock-ibm) [reply orderer/consensus/etcdraft/consenter.go]: yes, makes sense, I changed it to

"After eviction from the cluster Registrar.SwitchToFollower will be called, and the orderer will become a follower of the channel."

File: orderer/consensus/etcdraft/consenter.go (hunk 1 above)
Comment (tock-ibm) [reply orderer/consensus/etcdraft/consenter.go]: done

**Review by yacovm — APPROVED** [review]

**Pull request merged**